src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java

```diff
@@ -74,14 +76,22 @@ public void save( final URI identifier, final T element ) throws IOException {
 		}
 	}
 
+	private final String VERSION_CHARS = "[\\w\\.\\{\\}\\$]+";
+
 	@SuppressWarnings( "unchecked" )
 	@Override
 	public T load( final URI identifier ) throws IOException {
-		String writerVersion = "unknown";
+		String writerVersion = OLD_RECHECK_VERSION;
 		try ( final Input input = new Input( Files.newInputStream( Paths.get( identifier ) ) ) ) {
 			writerVersion = input.readString();
 			return (T) kryo.readClassAndObject( input );
 		} catch ( final KryoException e ) {
+			if ( version.equals( writerVersion ) ) {
```

Comment:
Optional and more of a matter of taste: maybe it is helpful to wrap those boolean expressions inside methods? For example:

```java
if ( isSameVersion( writerVersion ) ) {
    throw e;
}
if ( isUnknownFormat( writerVersion ) ) {
    writerVersion = OLD_RECHECK_VERSION;
}

Reply:
I think wrapping `version.equals(` is less readable, as it hides otherwise easily understandable code.